PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/ControlScript/MasterControl/DynamicThumbstick.rbxmx

```diff
@@ -0,0 +1,415 @@
+<roblox xmlns:xmime="http://www.w3.org/2005/05/xmlmime" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="http://www.roblox.com/roblox.xsd" version="4">
+	<External>null</External>
+	<External>nil</External>
+	<Item class="ModuleScript" referent="RBX1E3E090B922241488CF1527B6A17D7C7">
+		<Properties>
+			<string name="Name">DynamicThumbstick</string>
+			<ProtectedString name="Source"><![CDATA[--[[
+	// FileName: DynamicThumbstick
+	// Version 0.9
+	// Written by: jhelms
+	// Description: Implements dynamic thumbstick controls for touch devices
+--]]
+local Players = game:GetService('Players')
+local UserInputService = game:GetService('UserInputService')
+local GuiService = game:GetService('GuiService')
+local RunService = game:GetService('RunService')
+
+local MasterControl = require(script.Parent)
+
+local Thumbstick = {}
+
+--[[ Script Variables ]]--
+while not Players.LocalPlayer do
+	wait()
+end
+local LocalPlayer = Players.LocalPlayer
+
+local Backpack = LocalPlayer:WaitForChild("Backpack")
+local Tools = {}
+local IsAToolEquipped = false
+
+local IsPortraitMode = true
+
+local RevertAutoJumpEnabledToFalse = false
+
+local ThumbstickFrame = nil
+local ImageCenter = nil
+local ImageUnderThumb = nil
+local ImageRod = nil
+
+local IsFollowStick = false
+local ThumbstickFrame = nil
+local OnMoveTouchEnded = nil		-- defined in Create()
+local OnTouchMovedCn = nil
+local OnTouchEndedCn = nil
+local OnRenderSteppedCn
+local currentMoveVector = Vector3.new(0,0,0)
+
+--[[ Constants ]]--
+
+local IMAGE_OUTER_URL = "rbxasset://textures/ui/Disk.png"
+local IMAGE_STICK_URL = "rbxasset://textures/ui/Disk.png"
+local IMAGE_ROD_URL = "rbxasset://textures/ui/Square.png"
+
+local TOUCH_IS_TAP_TIME_THRESHOLD = 0.5
+local TOUCH_IS_TAP_DISTANCE_THRESHOLD = 25
+
+local FadeInAndOutBackground = true
+local FadeInAndOutMaxCount = nil
+local FadeInAndOutMaxAlpha = 0.35
+local FadeInAndOutHalfDuration = 0.3
+
+--[[ Local functionality ]]--
+
+local function adjustFrameSizeAndPosition()
+	if ThumbstickFrame then
+		if IsPortraitMode then
+			ThumbstickFrame.Size = UDim2.new(1, 0, .4, 0)
+			ThumbstickFrame.Position = UDim2.new(0, 0, 0.6, 0)
+		else
+			ThumbstickFrame.Size = UDim2.new(0.5, 0, 1, 0)
+			ThumbstickFrame.Position = UDim2.new(0, 0, 0, 0)
+		end
+	end
+end
+
+spawn(function()
+	while(true) do
+		--IsPortraitMode = newOrientation == Enum.ScreenOrientation.Portrait
```

Comment:
Actually, none of this section is needed! No need to track if portrait mode here , and this wouldn't be the way to do it. Meant to take it out.